public/app/features/correlations/types.ts

```diff
@@ -84,6 +88,8 @@ export type GetCorrelationsParams = {
   page: number;
 };
 
+export type OmitUnion<T, K extends keyof any> = T extends any ? Omit<T, K> : never;
```

Comment:
This is an issue with TS definition of the `Omit` type, https://github.com/microsoft/TypeScript/issues/31501. Basically it does not keep the discriminate union information which means we either have to manually write `export type CreateCorrelationParams = Omit<CorrelationQuery, 'uid' | 'provisioned'> | Omit<CorrelationExternal, 'uid' | 'provisioned'>;` to keep the discriminated union type or use this helper. This also translates to [CorrelationsPage.test.tsx](https://github.com/grafana/grafana/pull/92442/commits/a92b6e742fb10beb368aa9c49b5c47453006ec6f#diff-69c133d1806c9bba3cb538729cc389ee255c9c6dd1e530c661932cf9c7152d2e) where we also use `Omit` in a function param removing the discriminate union from the type and then trying to pass it to `Correlation` which is a discriminate union.